### DISCUSSION TITLE
optical force on particle placed inside nanobeam cavity

### DISCUSSION THREAD

**Discussion by TadeasManka** [NanoComp/meep#1953] [Q&A]
Hello, I would like to know your opinions no my simulation results. 
I have this device (very similar to holly_wvg_cavity in tutorials). We work in 2D, res = 60
![Snímek obrazovky 2022-02-17 v 13 14 41](https://user-images.githubusercontent.com/64955590/154480008-4dd09a5d-a0b6-42c2-9c7e-7ed9d3237bd3.png)
The only difference is that the main mode is not propagating in dielectric but in the air-holes...ignore slight difference in number of holes...
![Snímek obrazovky 2022-02-17 v 13 15 29](https://user-images.githubusercontent.com/64955590/154480163-6a223f18-aa43-4fcc-ad9c-449b6bdf97a1.png)
I placed the particle (infinite cylinder) inside the central hole and enclosed it with box of ForceRegions. 

`r_dist = 0.075`
`force_reg1 = mp.ForceRegion(center=mp.Vector3(x=+r_dist), direction=mp.X, weight=+1, size=mp.Vector3(y=2*r_dist))`
`force_reg2 = mp.ForceRegion(center=mp.Vector3(x=-r_dist), direction=mp.X, weight=-1, size=mp.Vector3(y=2*r_dist))`
`force_reg3 = mp.ForceRegion(center=mp.Vector3(y=+r_dist), direction=mp.Y, weight=+1, size=mp.Vector3(x=2*r_dist))`
`force_reg4 = mp.ForceRegion(center=mp.Vector3(y=-r_dist), direction=mp.Y, weight=-1, size=mp.Vector3(x=2*r_dist))`
        
`par_force = sim.add_force(fcen, df, nfreq, force_reg1, force_reg2, force_reg3, force_reg4)`

The particle (r = 0.05) as well as the box (r_dist = 0.075) are in the vacuum and the box is not touching the particle. I insert a Ey gaussian pulse on the left of the device and let it propagate to the right, where the fields are accumulated. The sim is stopped, when the fields decays by 1e-3. The transmission spectrum looks like this (transmission of device/transmission of normalization run), which is what I expected

![Snímek obrazovky 2022-02-17 v 13 22 42](https://user-images.githubusercontent.com/64955590/154481434-d9dc410d-fdc5-44a4-9713-52d4f80a230d.png)

On the other hand if we look on force spectrum (force/psd of only wvg) we get this:

![Snímek obrazovky 2022-02-17 v 13 23 16](https://user-images.githubusercontent.com/64955590/154481622-d2f4e089-9b64-49b3-8662-4f9a667d5285.png)

My question is, isn't this wrong? The force should be directly proportional to grad of intensity and the intensity at the center should be flat (max value but flat)...so I would expect no force at all at this equilibrium position. I would like to know your opinions...thanks.

**Comment by stevengj** [comment]
> The force should be directly proportional to grad of intensity

Gradient force is not the only kind of radiation pressure.

(e.g. planewave has constant intensity, but still exerts a force.)

**Reply by TadeasManka** [reply]
of course, I know there is scattering/radiation force and gradient force but in this example its not intuitive for me. I would expect, that it will be acting like optical tweezer (the fields look very similar to focused beam) and somehow pushing the particle to the center of the cavity. But instead the force is not zero and even negative. What does it mean? The particle will tend to move from the center?
